### PR TITLE
fix: normalization for companys/sponsors

### PR DESCRIPTION
**Pull request by Xetoxyc**
Improving the sponsor name handling to show the crown in stats

### CODE REVIEW

File: src/lib/participants/normalize-company.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	/\s+(?:ag|co\s*kg|e\.?\s*k|e\.?\s*v|eg|gbr|gmbh|gmdbh|inc|kg|ltd|llc|ohg|se|ug(?:\s*\(haftungsbeschr[aä]nkt\))?)\s*$/i;
+
+export function normalizeCompanyKey(name: string): string {
+	return name
+		.replace(LEGAL_SUFFIXES, '')
+		.toLowerCase()
+		.replace(/[^a-z0-9]/g, '-')
+		.replace(/-+/g, '-')
+		.replace(/^-|-$/g, '')
+		.replace(/^alm-engineering$/, 'alm');
```
Comment (Narigo) [inline src/lib/participants/normalize-company.ts:11]: 500€ line @michaelzoidl @dreampulse 😉

File: src/lib/participants/normalize-company.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const LEGAL_SUFFIXES =
+	/\s+(?:ag|co\s*kg|e\.?\s*k|e\.?\s*v|eg|gbr|gmbh|gmdbh|inc|kg|ltd|llc|ohg|se|ug(?:\s*\(haftungsbeschr[aä]nkt\))?)\s*$/i;
```
Comment (Narigo) [inline src/lib/participants/normalize-company.ts:2]: I guess there could be some additional dots for e.K. and e.V. - would be great to have unit tests for this... 😅

```suggestion
	/\s+(?:ag|co\s*kg|e\.?\s*k\.?|e\.?\s*v\.?|eg|gbr|gmbh|gmdbh|inc|kg|ltd|llc|ohg|se|ug(?:\s*\(haftungsbeschr[aä]e?nkt\))?)\s*$/i;
```

I think it won't match a GmbH & Co. KG right now

File: src/lib/participants/normalize-company.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export function normalizeCompanyKey(name: string): string {
+	return name
+		.replace(LEGAL_SUFFIXES, '')
+		.toLowerCase()
+		.replace(/[^a-z0-9]/g, '-')
```
Comment (Narigo) [inline src/lib/participants/normalize-company.ts:8]: I think we should replace ä, ö, ü, ß with ae, oe, ue, ss as well for the key

**Review by Narigo — APPROVED** [review]
Should work for the current sponsors, but maybe we want to add an additional PR to provide some additional test cases for the normalizeCompanyKey and filter the edge cases
